lib/sidekiq_flow/adapters/legacy_storage.rb

```diff
@@ -92,7 +92,7 @@ def already_started?(workflow_id)
       end
 
       def connection_pool
-        @connection_pool ||= ConnectionPool.new(size: configuration.concurrency, timeout: configuration.timeout) do
+        @connection_pool ||= configuration.connection_pool ? configuration.connection_pool : ConnectionPool.new(size: configuration.concurrency, timeout: configuration.timeout) do
```

Comment:
Move logic to `Configuration.setup_connection_pool`. We just want to see `@connection_pool ||= configuration.connection_pool` here.

See `SidekiqFlow.configure`.